### ISSUE TITLE
test: test_conf_change_remove_leader failed

### ISSUE THREAD

**Issue by zhangjinpeng1987** [tikv/tikv#4113]
```
test raftstore::test_conf_change::test_conf_change_remove_leader ... thread 'raftstore-3-0::test_conf_change_remove_leader' panicked at 'snap apply job should not fail: channel has been closed', src/libcore/result.rs:1009:5

note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

stack backtrace:

   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace

             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49

   1: std::sys_common::backtrace::_print

             at src/libstd/sys_common/backtrace.rs:71

   2: std::panicking::default_hook::{{closure}}

             at src/libstd/sys_common/backtrace.rs:59

             at src/libstd/panicking.rs:211

   3: std::panicking::default_hook

             at src/libstd/panicking.rs:227

   4: std::panicking::rust_panic_with_hook

             at src/libstd/panicking.rs:491

   5: std::panicking::continue_panic_fmt

             at src/libstd/panicking.rs:398

   6: rust_begin_unwind

             at src/libstd/panicking.rs:325

   7: core::panicking::panic_fmt

             at src/libcore/panicking.rs:95

   8: core::result::unwrap_failed

             at /rustc/14997d56a550f4aa99fe737593cd2758227afc56/src/libcore/macros.rs:26

   9: <core::result::Result<T, E>>::expect

             at /rustc/14997d56a550f4aa99fe737593cd2758227afc56/src/libcore/result.rs:835

  10: tikv::raftstore::store::peer_storage::PeerStorage::schedule_applying_snapshot

             at src/raftstore/store/peer_storage.rs:1062

  11: tikv::raftstore::store::peer_storage::PeerStorage::post_ready

             at src/raftstore/store/peer_storage.rs:1158

  12: tikv::raftstore::store::peer::Peer::post_raft_ready_append

             at ./src/raftstore/store/peer.rs:946

  13: <tikv::raftstore::store::fsm::peer::PeerFsmDelegate<'a, T, C>>::post_raft_ready_append

             at ./src/raftstore/store/fsm/peer.rs:527

  14: <tikv::raftstore::store::fsm::store::RaftPoller<T, C>>::handle_raft_ready

             at ./src/raftstore/store/fsm/store.rs:543

  15: <tikv::raftstore::store::fsm::store::RaftPoller<T, C> as tikv::raftstore::store::fsm::batch::PollHandler<tikv::raftstore::store::fsm::peer::PeerFsm, tikv::raftstore::store::fsm::store::StoreFsm>>::end

             at ./src/raftstore/store/fsm/store.rs:653

  16: <tikv::raftstore::store::fsm::batch::Poller<N, C, Handler>>::poll

             at ./src/raftstore/store/fsm/batch.rs:344

  17: <tikv::raftstore::store::fsm::batch::BatchSystem<N, C>>::spawn::{{closure}}

             at ./src/raftstore/store/fsm/batch.rs:411

thread 'raftstore::test_conf_change::test_conf_change_remove_leader' panicked at 'called `Result::unwrap()` on an `Err` value: Any', src/libcore/result.rs:1009:5

note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

stack backtrace:

   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace

             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49

   1: std::sys_common::backtrace::_print

             at src/libstd/sys_common/backtrace.rs:71

   2: std::panicking::default_hook::{{closure}}

             at src/libstd/sys_common/backtrace.rs:59

             at src/libstd/panicking.rs:211

   3: std::panicking::default_hook

             at src/libstd/panicking.rs:227

   4: std::panicking::rust_panic_with_hook

             at src/libstd/panicking.rs:491

   5: std::panicking::continue_panic_fmt

             at src/libstd/panicking.rs:398

   6: rust_begin_unwind

             at src/libstd/panicking.rs:325

   7: core::panicking::panic_fmt

             at src/libcore/panicking.rs:95

   8: core::result::unwrap_failed

             at /rustc/14997d56a550f4aa99fe737593cd2758227afc56/src/libcore/macros.rs:26

   9: <core::result::Result<T, E>>::unwrap

             at /rustc/14997d56a550f4aa99fe737593cd2758227afc56/src/libcore/result.rs:808

  10: <tikv::raftstore::store::fsm::batch::BatchSystem<N, C>>::shutdown

             at src/raftstore/store/fsm/batch.rs:429

  11: tikv::raftstore::store::fsm::store::RaftBatchSystem::shutdown

             at src/raftstore/store/fsm/store.rs:1115

  12: <tikv::server::node::Node<C>>::stop_store

             at ./src/server/node.rs:365

  13: <tikv::server::node::Node<C>>::stop

             at ./src/server/node.rs:372

  14: <test_raftstore::node::NodeCluster as test_raftstore::cluster::Simulator>::stop_node

             at components/test_raftstore/src/node.rs:271

  15: <test_raftstore::cluster::Cluster<T>>::stop_node

             at ./components/test_raftstore/src/cluster.rs:205

  16: <test_raftstore::cluster::Cluster<T>>::shutdown

             at ./components/test_raftstore/src/cluster.rs:512

  17: <test_raftstore::cluster::Cluster<T> as core::ops::drop::Drop>::drop

             at ./components/test_raftstore/src/cluster.rs:972

  18: core::ptr::real_drop_in_place

             at /rustc/14997d56a550f4aa99fe737593cd2758227afc56/src/libcore/ptr.rs:204

  19: integrations::raftstore::test_conf_change::test_conf_change_remove_leader

             at tests/integrations/raftstore/test_conf_change.rs:741

  20: integrations::raftstore::test_conf_change::test_conf_change_remove_leader::{{closure}}

             at tests/integrations/raftstore/test_conf_change.rs:718

  21: core::ops::function::FnOnce::call_once

             at /rustc/14997d56a550f4aa99fe737593cd2758227afc56/src/libcore/ops/function.rs:238

  22: <F as alloc::boxed::FnBox<A>>::call_box

             at src/libtest/lib.rs:1471

             at /rustc/14997d56a550f4aa99fe737593cd2758227afc56/src/libcore/ops/function.rs:238

             at /rustc/14997d56a550f4aa99fe737593cd2758227afc56/src/liballoc/boxed.rs:673

  23: __rust_maybe_catch_panic

             at src/libpanic_unwind/lib.rs:102
```

**Comment by zhangjinpeng1987** [comment]
Please check https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tikv_ghpr_test/detail/tikv_ghpr_test/1176/pipeline for more informations.

**Comment by overvenus** [comment]
Dup with #4101 and #4105.

**Comment by hicqu** [comment]
Closed with #4090 .
